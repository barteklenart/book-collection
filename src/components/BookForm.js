import React from 'react';

class BookForm extends React.Component {

	constructor(props){
		super(props);

		this.state = {
			title: '',
			description: '',
			completedAt: 0
		};
	}

	onTitleChange = (e) => {
		this.setState({
			title: e.target.value
		});
	}

	onDescriptionChange = (e) =>{
		this.setState({
			description: e.target.value
		})
	}

	onSubmit = (e) => {
		e.preventDefault();
		this.props.onSubmit({
			title: this.state.title,
			description: this.state.description,
			completedAt: this.state.completedAt
		})
	}

	render(){
		return (
			<form onSubmit={this.onSubmit}>
				<label htmlFor="title">Title:</label>
				<input 
					id="title" 
					type="text"
					value={this.state.title}
					onChange={this.onTitleChange} 
				/>
				<label htmlFor="content">Content</label>
				<textarea 
					id="content" 
					col="10" 
					row="5"
					value={this.state.content}
					onChange={this.onDescriptionChange}
				></textarea>
				<label>Completed</label>
				<button type="submit">Add Book</button>
			</form>
		);
	}
}

export default BookForm;