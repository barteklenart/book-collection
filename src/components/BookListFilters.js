import "react-dates/initialize";
import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';
import moment from 'moment';

import { setTextFilter, sortByDateUp, sortByDateDown, sortByTitleUp, sortByTitleDown, setStartDate, setEndDate } from '../actions/filters';

class BookListFilters extends React.Component {

	constructor(props) {
		super(props);
		this.state = {
			searchText: '',
			startDate: moment(),
			endDate: moment(),
			focusedInput: null
		}
	}

	onInputChange = (e) => {
		const searchText = e.target.value;
		this.setState({
			searchText
		});
		this.props.setTextFilter(searchText);
	}

	onSelect = (e) => {
		const { value } = e.target;

		switch(value){
			case 'dateup':
				this.props.sortByDateUp(value)
				break;
			case 'datedown':
				this.props.sortByDateDown(value)
				break;
			case 'titleup':
				this.props.sortByTitleUp(value);
				break;
			case 'titledown':
				this.props.sortByTitleDown(value);
		}
	}

	onDatesChange = ({ startDate, endDate }) => {
		this.setState({ startDate, endDate });
		this.props.setStartDate(startDate);
		this.props.setEndDate(endDate);
	}

	onFocusChange = (focusedInput) => {
		this.setState({ focusedInput });
	}

	render() {
		return (
			<div>
				<input
					type="text"
					value={this.state.searchText}
					onChange={this.onInputChange}
				/>
				<select onChange={this.onSelect}>
					<option value="dateup">Date ↑</option>
					<option value="datedown">Date ↓</option>
					<option value="titleup">Title ↑</option>
					<option value="titledown">Title ↓</option>
				</select>
				<DateRangePicker
					startDate={this.state.startDate} 
					startDateId="startDateId" 
					endDate={this.state.endDate} 
					endDateId="endDateId" 
					onDatesChange={this.onDatesChange} 
					focusedInput={this.state.focusedInput}
					onFocusChange={this.onFocusChange} 
				/>
			</div>
		);
	}
};

const mapDispatchToProps = (dispatch) => {
	return {
		setTextFilter: (text) => dispatch(setTextFilter(text)),
		sortByDateUp: (filter) => dispatch(sortByDateUp(filter)),
		sortByDateDown: (filter) => dispatch(sortByDateDown(filter)),
		sortByTitleUp: (filter) => dispatch(sortByTitleUp(filter)),
		sortByTitleDown: (filter) => dispatch(sortByTitleDown(filter)),
		setStartDate: (startDate) => dispatch(setStartDate(startDate)),
		setEndDate: (endDate) => dispatch(setEndDate(endDate))
	}
}

export default connect(undefined, mapDispatchToProps)(BookListFilters);