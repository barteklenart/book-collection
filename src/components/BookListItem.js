import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const BookListItem = ({ book, removeBook }) => {

	return (
		<div>
			<h3>{book.title}</h3>
			<p>{book.description}</p>
			<span>{moment(book.completedAt).format('MM-DD-YYYY')}</span>
			<Link to={`/edit/${book.id}`}>Edit</Link>
			<button onClick={() => removeBook(book.id)}>Remove</button>
			<hr />
		</div>
	);
};

export default BookListItem;