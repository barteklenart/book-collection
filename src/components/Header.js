import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => {
	return (
		<header>
			<h1>Book collection</h1>			
			<NavLink to="/" activeClassName="current">Dashboard</NavLink>
			<NavLink to="/edit" activeClassName="current">Edit Book</NavLink>
			<NavLink to="/add" activeClassName="current">Add Book</NavLink>
		</header>
	);
};

export default Header;