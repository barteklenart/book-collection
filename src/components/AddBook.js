import React from 'react';
import { connect } from 'react-redux';
import { addBook } from '../actions/books';

import BookForm from './BookForm';

const AddBook = (props) => {
	const onSubmit = (book) => {
		props.addBook(book);
	}
	return (
		<div>
			<h2>Add new book</h2>
			<BookForm onSubmit={onSubmit}/>
		</div>
	);
};

const mapDispatchToProps = (dispatch) => {
	return {
		addBook: (book) => dispatch(addBook(book))
	};
};

export default connect(undefined, mapDispatchToProps)(AddBook);