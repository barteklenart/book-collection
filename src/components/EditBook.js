import React from 'react';
import { connect } from 'react-redux';
import { editBook } from '../actions/books';

import BookForm from './BookForm';


class EditBook extends React.Component {
	onSubmit = (book) => {
		this.props.editBook(this.props.book.id, book)
		this.props.history.push('/');
	}

	render(){
		return (
			<div>
				<BookForm onSubmit={this.onSubmit} book={this.props.book}/>
			</div>
		);
	}
}

const mapStateToProps = (state, props) => {
	return {
		book: state.books.find((el) => el.id === props.match.params.id)
	};
};

const mapDispatchToProps = (dispatch, props) => ({
	editBook: (id, book) => dispatch(editBook(id, book))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditBook);