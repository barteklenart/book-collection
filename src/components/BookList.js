import React from 'react';
import { connect } from 'react-redux';

import BookListItem from './BookListItem';
import { removeBook } from '../actions/books';
import selectBook from '../selectors/book';

const BookList = (props) => {
	const removeBook = (id) => {
		props.removeBook(id);
	};

	return (
		<div>
			{props.books.length === 0 ? <p>No books</p> : props.books.map((book) => { return <BookListItem key={book.id} book={book} removeBook={removeBook} /> })}
		</div>
	);
}

const mapStateToProps = (state) => {
	return {
		books: selectBook(state.books, state.filters)
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		removeBook: (id) => dispatch(removeBook(id))	
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(BookList);