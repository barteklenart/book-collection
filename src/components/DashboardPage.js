import React from 'react';
import Header from './Header';

import BookList from './BookList';
import BookListFilters from './BookListFilters';

const DashboardPage = () => {
	return (
		<div>
			<BookListFilters />
			<BookList />
		</div>
	);
};

export default DashboardPage;