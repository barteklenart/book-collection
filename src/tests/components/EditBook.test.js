import React from 'react';
import { shallow } from 'enzyme';

import { EditBook } from '../../components/EditBook';

test('should render EditBook correctly', () => {
	const wrapper = shallow(<EditBook />);
	expect(wrapper).toMatchSnapshot();
});
