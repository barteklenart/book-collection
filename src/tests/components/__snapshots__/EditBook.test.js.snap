// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditBook correctly 1`] = `
<div>
  <BookForm
    onSubmit={[Function]}
  />
</div>
`;
